tests/baselines/reference/intersectionReduction.errors.txt

```diff
@@ -44,6 +47,11 @@ tests/cases/conformance/types/intersection/intersectionReduction.ts(81,1): error
     ab.kind;  // Error
        ~~~~
 !!! error TS2339: Property 'kind' does not exist on type 'never'.
+!!! error TS2339:   The type 'never' was reduced from the intersection 'A & B'. Each type of that intersection has properties that conflict, so values of that type can never exist.

```

Comment:
You should probably search for and report the properties that conflict and their respective types, no? So the error is somewhat more actionable?

Reply:
Definitely seems doable, but only if the approach in this PR seems reasonable.

Reply:
Seems OK. We could probably consider leaving behind more "breadcrumbs" like this in the future to explain how a specific type came about; but it's always about striking that balance between useful explaianation and terseness.